### DISCUSSION TITLE
Thanos sidecar, store and query enable TLS on HTTP protocol

### DISCUSSION THREAD

**Discussion by selvamurugan-altnix** [thanos-io/thanos#3754] [General]
Is there any method or workaround to enable TLS on HTTP protocol for Thanos sidecar, store, and query?
Because I am unable to get any solution and documentation if anyone did the configuration or solution, please help me out.

**Comment by GiedriusS** [comment]
Hi, Thanos components by themselves don't support TLS because it is very easy to put a reverse proxy in front that could do just that, and have many more features that it wouldn't be rational to implement in Thanos itself. Some popular options include Caddy, nginx, traefik. Have you tried looking into any of those?
